lgc/patch/PatchCheckShaderCache.cpp

```diff
@@ -89,8 +89,8 @@ bool PatchCheckShaderCache::runOnModule(Module &module) {
   Patch::init(&module);
 
   // NOTE: Global constants are either added to the end of the .text section, or in a separate .rodata section with
-  // relocs in the .text section to refer to them. We can't merge ELF binaries if relocs are used because llpcElfWriter
-  // doesn't know how to merge them.
+  // relocs in the .text section to refer to them. By enhancing llpcElfWriter for relocs, we can merge the relocs
+  // used by the fragment shader.
```

Comment:
In shouldEmitConstantsToTextSection() in llvm/lib/Target/AMDGPU/Utils/AMDGPUBaseInfo.cpp, it looks to me like constants are always in a separate .rodata section with relocs in .text. So we should be able to cope with any global data, not just global data in the fragment shader, right?

Reply:
The code you mentioned also confuses me. During the implementation of this change, I found that the constants are always appended to the end of the .text section. You can reference the attached pipeline.
[reloc_pipeline.zip](https://github.com/GPUOpen-Drivers/llpc/files/6690751/reloc_pipeline.zip)

Reply:
Seems like constants are currently forced into the .text section here: https://github.com/GPUOpen-Drivers/llpc/blob/312d5e625f5d0b360dd5190095c771693388888d/lgc/patch/PatchPreparePipelineAbi.cpp#L400-L403
Commenting out that line moves the constants into a .rodata section for that pipeline.

Reply:
When do you think we can default enable the .rodata section for relocs? This change is based on the current driver behavior. Do you think we need additional support for the .rodata section?

Reply:
I believe we can switch to .rodata now. You could do that as part of this change. @Flakebi, do you agree?

Reply:
Yes, I agree with that. Removing `PatchPreparePipelineAbi::setConstantGlobalSection` seems good to go.

Reply:
Sure, will do.